processor/src/main/java/org/jboss/logging/processor/generator/model/ClassModel.java

```diff
@@ -362,4 +368,26 @@ private JExpr determineLocale(final String locale, final JType localeType) {
 
         return initializer;
     }
+
+    private static class JFilerOriginatingElementAware extends JFiler {
+
+        private final Element originatingElement;
+        private final Filer filer;
+
+        private JFilerOriginatingElementAware(Element originatingElement, Filer filer) {
+            this.originatingElement = originatingElement;
```

Comment:
I'd check here for these elements to not be null?
Otherwise it fails later and it would get trickier to narrow down

Reply:
I'm going to propose this to happen slightly differently. I'd propose throwing a `ProcessingException` as it can take the element as an argument and potentially fail with a better message.

Reply:
ok 😃, so maybe something like this ?

Reply:
Perfect. I know these are highly unlikely, but may as well have as much information as possible if they are thrown :)